stock-filters/ChirpVillage/Builder.py

```diff
@@ -0,0 +1,263 @@
+    elif face == 1:
+        for i in range(z, z+2):
+            for j in range(y, y-2, -1):
+                utilityFunctions.setBlock(
+                    level, window_block, x, j, i)
+
+
+"""
+    coords must be top block
+    generates window like:
+    #
+    ###
+    #
+"""
+def construct_circle_window(level, coords, biome, face=0):
+    x = coords[0]
+    y = coords[1]
+    z = coords[2]
+    window_block = BlockUtils.get_window_block(biome)
+    if face == 0:
+        for j in range(y, y-3, -1):
+            utilityFunctions.setBlock(
+                level, window_block, x, j, z)
+            if j == y-1:
+                utilityFunctions.setBlock(
+                    level, window_block, x-1, j, z)
+                utilityFunctions.setBlock(
+                    level, window_block, x+1, j, z)
+    elif face == 1:
+        for j in range(y, y-3, -1):
+            utilityFunctions.setBlock(
+                level, window_block, x, j, z)
+            if j == y-1:
+                utilityFunctions.setBlock(
+                    level, window_block, x, j, z-1)
+                utilityFunctions.setBlock(
+                    level, window_block, x, j, z+1)
+
+
+"""
+    coords must be top block
+    generates window like:
+    #
+    ###
+"""
+def construct_triangle_window(level, coords, biome, face=0):
+    x = coords[0]
+    y = coords[1]
+    z = coords[2]
+    window_block = BlockUtils.get_window_block(biome)
+    if face == 0:
+        for j in range(y, y-2, -1):
+            utilityFunctions.setBlock(
+                level, window_block, x, j, z)
+            if j == y-1:
+                utilityFunctions.setBlock(
+                    level, window_block, x-1, j, z)
+                utilityFunctions.setBlock(
+                    level, window_block, x+1, j, z)
+    elif face == 1:
+        for j in range(y, y-2, -1):
+            utilityFunctions.setBlock(
+                level, window_block, x, j, z)
+            if j == y-1:
+                utilityFunctions.setBlock(
+                    level, window_block, x, j, z-1)
+                utilityFunctions.setBlock(
+                    level, window_block, x, j, z+1)
+
+
+def remove_blocks_in_box(level, coords):
+    minx = coords[0][0]
+    minz = coords[0][1]
+    miny = coords[0][2]
+    maxx = coords[1][0]
+    maxz = coords[1][1]
+    maxy = coords[1][2]
+    # Sets every block in box to air
+    for y in range(miny, maxy):
+        for x in range(minx, maxx):
+            for z in range(minz, maxz):
+                utilityFunctions.setBlock(level, (0, 0), x, y, z)
+
+
+"""Building Classes"""
+class BasicBuilding:
+
+    def construct(self, level, coords, surface):
+        # This is assumes box is already the size of the house
+        minx = surface.to_real_x(coords[0][0])
+        minz = surface.to_real_z(coords[0][1])
+        maxx = surface.to_real_x(coords[1][0])
+        maxz = surface.to_real_z(coords[1][1])
+        block = surface.surface_map[minx][minz]
```

Comment:
minx and minz are real coords so shouldnt work on surface_map. You may be able to replace them with coords[0][0] and coords[0][1]

Reply:
In this case I want real coords so that I can place the blocks in the world.

Reply:
the to_real methods are meant to convert a surface map coord to a real level coord

Reply:
I misread the comment, you're right